aws/provider_test.go

```diff
@@ -355,3 +357,212 @@ func testSweepSkipSweepError(err error) bool {
+	for _, endpointServiceName := range endpointServiceNames {
+		// Skip deprecated endpoint configurations as they will override expected values
+		if endpointServiceName == "kinesis_analytics" || endpointServiceName == "r53" {
+			continue
+		}
+
+		endpoints.WriteString(fmt.Sprintf("%s = \"http://%s\"\n", endpointServiceName, endpointServiceName))
+	}
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:          func() { testAccPreCheck(t) },
+		ProviderFactories: testAccProviderFactories(&providers),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAWSProviderConfigEndpoints(endpoints.String()),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAWSProviderEndpoints(&providers),
+				),
+			},
+		},
+	})
+}
+
+func TestAccAWSProvider_Endpoints_Deprecated(t *testing.T) {
+	var providers []*schema.Provider
+	var endpointsDeprecated strings.Builder
+
+	// Initialize each deprecated endpoint configuration with matching name and value
+	for _, endpointServiceName := range endpointServiceNames {
+		// Only configure deprecated endpoint configurations
+		if endpointServiceName != "kinesis_analytics" && endpointServiceName != "r53" {
+			continue
+		}
+
+		endpointsDeprecated.WriteString(fmt.Sprintf("%s = \"http://%s\"\n", endpointServiceName, endpointServiceName))
+	}
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:          func() { testAccPreCheck(t) },
+		ProviderFactories: testAccProviderFactories(&providers),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAWSProviderConfigEndpoints(endpointsDeprecated.String()),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAWSProviderEndpointsDeprecated(&providers),
+				),
+			},
+		},
+	})
+}
+
+func testAccCheckAWSProviderEndpoints(providers *[]*schema.Provider) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		if providers == nil {
+			return fmt.Errorf("no providers initialized")
+		}
+
+		// Match AWSClient struct field names to endpoint configuration names
+		endpointFieldNameF := func(endpoint string) func(string) bool {
+			return func(name string) bool {
+				switch endpoint {
+				case "applicationautoscaling":
+					endpoint = "appautoscaling"
+				case "budgets":
+					endpoint = "budget"
+				case "cloudformation":
+					endpoint = "cf"
+				case "cloudhsm":
+					endpoint = "cloudhsmv2"
+				case "cognitoidentity":
+					endpoint = "cognito"
+				case "configservice":
+					endpoint = "config"
+				case "cur":
+					endpoint = "costandusagereport"
+				case "directconnect":
+					endpoint = "dx"
+				case "lexmodels":
+					endpoint = "lexmodel"
+				case "route53":
+					endpoint = "r53"
+				case "sdb":
+					endpoint = "simpledb"
+				case "serverlessrepo":
+					endpoint = "serverlessapplicationrepository"
+				case "servicecatalog":
+					endpoint = "sc"
+				case "servicediscovery":
+					endpoint = "sd"
+				case "stepfunctions":
+					endpoint = "sfn"
+				}
+
+				switch name {
```

Comment:
Thought about recommending an if statement here, but the multiple `||` is too much, and I don't feel strongly enough to say use a regex match :100: